pkg/artifactory/resource_artifactory_local_debian_repository.go

```diff
@@ -4,53 +4,54 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-var debianLocalSchema = mergeSchema(baseLocalRepoSchema, map[string]*schema.Schema{
-	"primary_keypair_ref": {
-		Type:        schema.TypeString,
-		Optional:    true,
-		Description: "Used to sign index files in Debian artifacts. ",
-	},
-	"secondary_keypair_ref": {
-		Type:        schema.TypeString,
-		Optional:    true,
-		Description: "Used to sign index files in Debian artifacts. ",
-	},
-	"trivial_layout": {
-		Type:        schema.TypeBool,
-		Optional:    true,
-		Description: "When set, the repository will use the deprecated trivial layout.",
-		Deprecated:  "You shouldn't be using this",
-	},
-}, compressionFormats)
-
 func resourceArtifactoryLocalDebianRepository() *schema.Resource {
+	const pkt = "debian"
+
+	var debianLocalSchema = mergeSchema(getBaseLocalRepoSchema(pkt), map[string]*schema.Schema{
+		"primary_keypair_ref": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Used to sign index files in Debian artifacts. ",
+		},
+		"secondary_keypair_ref": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Used to sign index files in Debian artifacts. ",
+		},
+		"trivial_layout": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "When set, the repository will use the deprecated trivial layout.",
+			Deprecated:  "You shouldn't be using this",
+		},
+	}, compressionFormats)
+
+	type DebianLocalRepositoryParams struct {
+		LocalRepositoryBaseParams
+		TrivialLayout           bool     `hcl:"trivial_layout" json:"debianTrivialLayout,omitempty"`
+		IndexCompressionFormats []string `hcl:"index_compression_formats" json:"optionalIndexCompressionFormats,omitempty"`
+		PrimaryKeyPairRef       string   `hcl:"primary_keypair_ref" json:"primaryKeyPairRef,omitempty"`
+		SecondaryKeyPairRef     string   `hcl:"secondary_keypair_ref" json:"secondaryKeyPairRef,omitempty"`
+	}
+
+	var unPackLocalDebianRepository = func(data *schema.ResourceData) (interface{}, string, error) {
+		d := &ResourceData{ResourceData: data}
+		repo := DebianLocalRepositoryParams{
+			LocalRepositoryBaseParams: unpackBaseRepo("local", data, "debian"),
```

Comment:
You can use your constant here

Reply:
variable packageType used to replace litteral `debian` from all occurrences